### PR TITLE
Fix issue with read offset being ignored

### PR DESCRIPTION
**Pull request by alamar**
Add tests
see OpenHFT/Chronicle-Core#602

### CODE REVIEW

**Comment by alamar** [comment]
@peter-lawrey @JerryShea WDYT?

File: src/test/java/net/openhft/chronicle/bytes/readme/PrimitiveTest.java [hunk 1]
```diff
@@ -221,9 +221,22 @@ public void testBinaryPrimitiveOffset() {
     }
 
     @Test
-    public void testTextPrimitive() {
+    public void testTextPrimitiveByteBuffer() {
+        doTestTextPrimitive(Bytes.elasticHeapByteBuffer(64));
```
Comment (JerryShea) [inline src/test/java/net/openhft/chronicle/bytes/readme/PrimitiveTest.java:225]: these tests do not break if I revert the change HeapBytesStore

**Review by JerryShea — APPROVED** [review]

File: src/test/java/net/openhft/chronicle/bytes/HeapByteStoreTest.java [hunk 2]
```diff
@@ -48,4 +53,35 @@ public void testElasticBytesEnsuringCapacity() {
...
+    public void testWriteOnHeap() throws Exception {
+        doTestWrite(() -> ByteBuffer.allocate(128));
+    }
+
+    @Test
+    public void testWriteDirect() throws Exception {
+        doTestWrite(() -> ByteBuffer.allocateDirect(128));
+    }
+
+    private void doTestWrite(Callable<ByteBuffer> generator) throws Exception {
```
Comment (JerryShea) [inline src/test/java/net/openhft/chronicle/bytes/HeapByteStoreTest.java]: consider moving this test to BytesTest so it tests this behaviour is checked for all Bytes types

File: src/test/java/net/openhft/chronicle/bytes/readme/PrimitiveTest.java (hunk 1 above)
Comment (alamar) [reply src/test/java/net/openhft/chronicle/bytes/readme/PrimitiveTest.java:225]: They do, but I still think more code coverage do not hurt. They used to be broken by something else which since have been taken care of, so I believe.

Comment (JerryShea) [reply src/test/java/net/openhft/chronicle/bytes/readme/PrimitiveTest.java:225]: Fair enough!

**Review by JerryShea — APPROVED** [review]

**Comment by JerryShea** [comment]
This is a great catch
